Add grey fill in most of the charts

LeoLuosifen opened the pull request.

Sallie00 committed:
fix(test): fix test_summary.py

LeoLuosifen committed:
feat: updated that most of the charts can be grey fill

LeoLuosifen committed:
fix: fixed test file and updated the psy-chart

LeoLuosifen committed:
fix: updated the python.yml that does not need to test in all branches

ziqi0067 commented:
@FedericoTartarini We have finshed all tasks and passed the tests. Please review it.

FedericoTartarini commented:
@LeoLuosifen great work. However, I found the following issue

1. chart breaking - please fix it

<img width="1852" height="598" alt="image" src="https://github.com/user-attachments/assets/d6dad003-516e-448c-9265-0235cbce4702" />

2. wrong legend colors - please fix it

<img width="245" height="166" alt="image" src="https://github.com/user-attachments/assets/024b6728-2271-4336-be32-56d70a36a7d5" />

3. strange layout - I am not sure what to suggest here and in the UTCI chart since they are hard to interpret when we filter by hour. Perhaps we cannot implement the gray out feature here
  
<img width="476" height="596" alt="image" src="https://github.com/user-attachments/assets/04ed3912-f130-42c7-af5a-14c969205140" />

4. dots are shown - when they should not be - avoid implementing the gray out feature here

<img width="708" height="696" alt="image" src="https://github.com/user-attachments/assets/dc17f0ef-b818-459b-a22a-091ac1f7e6ab" />

5. strage behaviour - see comment 4

<img width="561" height="484" alt="image" src="https://github.com/user-attachments/assets/b7b1ed51-c258-474c-bde1-0d95a5cebd9c" />

FedericoTartarini commented:
@LeoLuosifen the new code is great but there are so many repetitions, can we try to avoid repetitions, if possible otherwise the code is going to be very hard to maintain.

giobetti commented:
Dear @Federico and @LeoLuosifen<https://github.com/LeoLuosifen> , thanks for all the great changes!
If I can interject here, I would suggest that the grey out feature should not be part of the bar charts (i.e. Cloud cover and UTCI). They should simply show the summary statistics with the filtered data.
Best,

Giovanni
Message ID: ***@***.***>

LeoLuosifen committed:
fix: fixed the chart's show and optimized code.
- fixed the yearly chart breaking
- fixed the legend color
- removed the grey fill in Cloud coverage, Psychrometric Chart, and the UTCI chart
- extracted the grey fill function

LeoLuosifen commented:
@FedericoTartarini @giobetti Thanks for your review and suggestions. I've already tried to resolve the issues mentioned above.

LeoLuosifen committed:
fix: optimized the duplicated code

FedericoTartarini committed:
fix: optimize test execution by enabling parallel processing

FedericoTartarini committed:
fix: test run 2 in parallel

FedericoTartarini committed:
fix: increase test parallelism and add readiness check for Clima

FedericoTartarini committed:
fix: remove parallel processing from Clima test execution

FedericoTartarini commented:
@LeoLuosifen amazing work. I have reviewed the changes and I am happy with the code. I am going to publish a new test version. I also discovered how to run the tests in parallel locally, but it does not work in the github action `pipenv run pytest --base-url=http://127.0.0.1:8080 -vv --numprocesses 4`
